dronegen/windows.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import "path"
+
+const (
+	perBuildWorkspace = `$Env:WORKSPACE_DIR/$Env:DRONE_BUILD_NUMBER`
+	toolchainDir      = `/toolchains`
+	teleportSrc       = `/go/src/github.com/gravitational/teleport`
+	webappsSrc        = `/go/src/github.com/gravitational/webapps`
+)
+
+func newWindowsPipeline(name string) pipeline {
+	p := newExecPipeline(name)
+	p.Workspace.Path = path.Join("C:/Drone/Workspace", name)
+	p.Concurrency.Limit = 1
+	p.Platform = platform{OS: "windows", Arch: "amd64"}
+	return p
+}
+
+func windowsTagPipeline() pipeline {
+	p := newWindowsPipeline("build-native-windows-amd64")
+
+	//p.Trigger = triggerTag
+	p.Trigger = trigger{
+		Event:  triggerRef{Include: []string{"push"}, Exclude: []string{"pull_request"}},
+		Branch: triggerRef{Include: []string{"master", "branch/*", "tcsc/build-windows*"}},
+		Repo:   triggerRef{Include: []string{"gravitational/*"}},
+	}
+
+	// TODO(tcsc): restore before merge
+	//p.DependsOn = []string{ "build-windows-amd64" }
+
+	p.Steps = []step{
+		cloneWindowsRepositoriesStep(p.Workspace.Path),
+		updateWindowsSubreposStep(p.Workspace.Path),
+		installWindowsNodeToolchainStep(p.Workspace.Path),
```

Comment:
After we get through this initial release, we should revisit this. Maybe now that we have automation for standing up the builders we can start to bundle more of the toolchains into the image instead of installing per-build.